src/Gaffer/FileSystemPath.cpp

```diff
@@ -578,6 +578,17 @@ void FileSystemPath::rootAndNames(const std::string &string, InternedString &roo
 {
 	std::string sanitizedString = string;
 
+	// Parsing a filename with characters that are meant to be escaped by string substitution will cause
+	// unwanted path splits. We replace Gaffer's escaped special characters with characters that can't be
+	// used in Windows paths (so they won't be expected to be part of the input), then convert back after
+	// splitting the path.
+	using ReplacementPair = std::pair<std::string, std::string>;
+	std::vector<ReplacementPair> replacements = { { "\\~", "<" }, { "\\$", ">" }, { "\\#", "?" } };
```

Comment:
Probably worth making this const and static, rather than allocating a temporary vector for every call to `rootAndNames`?